### PR TITLE
fix: scala3 completion - fix isses with code indented using tabs

### LINKED ISSUE #3698: Autocomplete not working properly for tab-indented Scala 3 code
**Describe the bug**

Intellisense/autocomplete does not work as expected in Scala 3 projects on the lines that are indented with tabs. 

**To Reproduce** Steps to reproduce the behavior:

1. Open a Scala 3 project
2. Indent a line of code with one or more tabs
3. Anywhere within the line, add a dot after an identifier, and press Ctrl+Space
4. "No suggestions" pops up instead of the expected list.

**Expected behavior**

Expected to work like for space-indented lines.

**Interesting workaround**

Adding a number of spaces after the dot, equal to the number of indentation tabs, helps (see the second screenshot below).

**Screenshots**

Not working:

scala_vscode_autocomp_not_working

Working:

scala_vscode_autocomp_working

**Installation:**

- Operating system: Linux
- VSCode version: 1.65.1
- VSCode extension version: 1.13.0
- Metals version: 0.11.2

**Additional context**

There are those who prefer indenting their code with tabs (at the beginning of the lines), aligning with spaces... Please, support us as well!

**Search terms**

VSCode, Metals, Scala 3, tabs, indentation, intellisense, autocomplete, not working

### PR DESCRIPTION
**Pull request by dos65**
Fixes #3698

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionPos.scala [hunk 1]
```diff
@@ -25,12 +25,12 @@ case class CompletionPos(
 
   def toEditRange: l.Range =
     def toPos(offset: Int): l.Position =
-      new l.Position(
-        cursorPos.source.offsetToLine(offset),
-        cursorPos.source.column(offset)
-      )
+      val line = cursorPos.source.offsetToLine(offset)
+      val column = offset - line
+      new l.Position(line, column)
```
Comment (kpodsiad) [inline mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionPos.scala:30]: Could you explain what was wrong? I cannot comprehend this by myself :D

Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionPos.scala:30]: This looks a bit weird. Why would be subtracting line number from file offset :thinking:

**Review by tgodzik — APPROVED** [review]
It looks good now! Thanks!

File: mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionPos.scala (hunk 1 above)
Comment (dos65) [reply mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionPos.scala]: oops. updated

@kpodsiad  the actual issue is that dotty's `SourceFile.column` method treats tabs incorrectly. If line starts with tabs they just don't count as symbols that results to having a wrong editRange in completionItems.

I will look how to fix that in dotty but we anyway need to fix in Metals for existing scala3 versions.

**Pull request merged**